packages/ddex/ingester/parser/parser.go

```diff
@@ -61,6 +72,93 @@ func RunNewParser(ctx context.Context) {
 	}
 }
 
+// createArtistNameIndex creates an index on the name (display name) field in the 'users' collection
+func (p *Parser) createArtistNameIndex() error {
+	_, err := p.usersColl.Indexes().CreateOne(p.ctx, mongo.IndexModel{
+		Keys: bson.M{"name": 1},
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (p *Parser) getArtistID(artistName string) (string, error) {
+	// Search Mongo for an exact match on artist name
+	filter := bson.M{"name": artistName}
+	cursor, err := p.usersColl.Find(p.ctx, filter)
+	if err != nil {
+		return "", err
+	}
+	defer cursor.Close(p.ctx)
+
+	var results []bson.M
+	if err = cursor.All(p.ctx, &results); err != nil {
+		return "", err
+	}
+
+	// Fail if multiple artists have their display name set to the same thing
+	if len(results) > 1 {
+		var matchingHandles []string
```

Comment:
nit: this var name is a little confusing bc the handles should be different; it's the display names that are the same, yea?

Reply:
yeah it's more like `handlesWithMatchingDisplayName` which is gross, so i'll change it to `handles`